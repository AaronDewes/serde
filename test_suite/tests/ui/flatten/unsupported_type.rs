use serde_derive::Deserialize;

#[derive(Deserialize)]
struct Outer {
    outer: String,
    #[serde(flatten)]
    inner: String,
}

fn main() {}
