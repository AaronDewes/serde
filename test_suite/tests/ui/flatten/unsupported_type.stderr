error[E0277]: the trait bound `std::string::String: DeserializeEmbed<'de>` is not satisfied
 --> $DIR/unsupported_type.rs:3:10
  |
3 | #[derive(Deserialize)]
  |          ^^^^^^^^^^^ the trait `DeserializeEmbed<'de>` is not implemented for `std::string::String`
  |
  = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: DeserializeEmbed<'de>` is not satisfied in `__Key<'de>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^ within `__Key<'de>`, the trait `DeserializeEmbed<'de>` is not implemented for `std::string::String`
    |
   ::: $WORKSPACE/serde/src/de/mod.rs
    |
    | pub trait Deserialize<'de>: Sized {
    |                             ----- required by this bound in `Deserialize`
    |
note: required because it appears within the type `__Key<'de>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^
    = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: DeserializeEmbed<'de>` is not satisfied in `__Key<'de>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^ within `__Key<'de>`, the trait `DeserializeEmbed<'de>` is not implemented for `std::string::String`
    |
   ::: $RUST/core/src/result.rs
    |
    | pub enum Result<T, E> {
    |                 - required by this bound in `Result`
    |
note: required because it appears within the type `__Key<'de>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^
    = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: DeserializeEmbed<'_>` is not satisfied in `__Key<'_>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^ within `__Key<'_>`, the trait `DeserializeEmbed<'_>` is not implemented for `std::string::String`
    |
   ::: $RUST/core/src/result.rs
    |
    | pub enum Result<T, E> {
    |                 - required by this bound in `Result`
    |
note: required because it appears within the type `__Key<'_>`
   --> $DIR/unsupported_type.rs:3:10
    |
3   | #[derive(Deserialize)]
    |          ^^^^^^^^^^^
    = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
